File: onebot/src/main/java/cn/evole/onebot/sdk/response/group/GroupMemberInfoResp.java
package cn.evole.onebot.sdk.response.group;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Created on 2022/7/8.
 *
 * @author cnlimiter
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class GroupMemberInfoResp {

    @SerializedName("group_id")
    public long groupId;

    @SerializedName("user_id")
    public long userId;

    @SerializedName("nickname")
    public String nickname;

    @SerializedName("card")
    public String card;

    @SerializedName("sex")
    public String sex;

    @SerializedName("age")
    public int age;

    @SerializedName("area")
    public String area;

    @SerializedName("join_time")
    public int joinTime;

    @SerializedName("last_sent_time")
    public int lastSentTime;

    @SerializedName("level")
    public int level;

    @SerializedName("role")
    public String role;

    @SerializedName("unfriendly")
    public boolean unfriendly;

    @SerializedName("title")
    public String title;

    @SerializedName("title_expire_time")
    public long titleExpireTime;

    @SerializedName("card_changeable")
    public boolean cardChangeable;

    @SerializedName("shut_up_timestamp")
    public long shutUpTimestamp;

}

File: overflow-core/src/main/kotlin/cn/evolvefield/onebot/sdk/response/group/GroupMemberInfoResp.kt
package cn.evolvefield.onebot.sdk.response.group

import com.google.gson.annotations.SerializedName

class GroupMemberInfoResp {
    @SerializedName("group_id")
    var groupId = 0L
    @SerializedName("user_id")
    var userId = 0L
    @SerializedName("nickname")
    var nickname = ""
    @SerializedName("card")
    var card = ""
    @SerializedName("sex")
    var sex = ""
    @SerializedName("age")
    var age = 0
    @SerializedName("area")
    var area = ""
    @SerializedName("join_time")
    var joinTime = 0
    @SerializedName("last_sent_time")
    var lastSentTime = 0
    @SerializedName("level")
    var level = 0
    @SerializedName("qq_level")
    var qqLevel = 0
    @SerializedName("role")
    var role = "member"
    @SerializedName("unfriendly")
    var unfriendly = false
    @SerializedName("title")
    var title = ""
    @SerializedName("title_expire_time")
    var titleExpireTime = 0L
    @SerializedName("card_changeable")
    var cardChangeable = true
    @SerializedName("shut_up_timestamp")
    var shutUpTimestamp = 0L
}
